src/modules/main.rs

```diff
@@ -71,7 +71,7 @@ impl TranslateModule for Main {
             let dollar = meta.gen_dollar();
             let args = self.args.clone().map_or_else(
                 String::new,
-                |name| format!("{name}=({quote}{dollar}0{quote} {quote}{dollar}@{quote})")
+                |name| format!("{name}=({quote}{dollar}@{quote})")
```

Comment:
This is the one line change for #564.

Reply:
Backed out this change; @Ph0enixKM points out that this was a deliberate change requested in #357.

Reply:
maybe we can add a parameter to define this behavior?

Reply:
I think what would cause more trouble than it's worth, tbh.

Reply:
but if we are readding it maybe it is something that we need to decide or in the future we are doing it again.

Reply:
No, I take @Ph0enixKM's point that Amber should replicate Bash `$@` functionality, and include the program or script name in the array passed to the `main` block. My problem is that a `for arg in "$@"` loop in native Bash skips over the first item in `$@`:
```
$ cat args.sh
#!/usr/bin/env bash

# Skips first item when iterating over "$@".
echo "First command line '$0'"
for arg in "$@"; do
    echo "Looping command line '$arg'"
done

# Includes first item when iterating over local array.
array=('1' '2' '3')
for arg in "${array[@]}"; do
    echo "Looping local array '$arg'"
done
```
```
$ ./args.sh 1 2 3
First command line './args.sh'
Looping command line '1'
Looping command line '2'
Looping command line '3'
Looping local array '1'
Looping local array '2'
Looping local array '3'
```